lib/sms.js

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const libpn = require('google-libphonenumber')
+const Nexmo = require('nexmo')
+const P = require('bluebird')
+const SENDER_IDS = new Map(require('./sms-sender-ids.json'))
+
+module.exports = (log, translator, templates, config) => {
+  const nexmo = new Nexmo({
+    apiKey: config.apiKey,
+    apiSecret: config.apiSecret
+  })
+  const sendSms = P.promisify(nexmo.message.sendSms, { context: nexmo.message })
+
+  // TODO: Sort out the real link
+  const LINK = 'https://moz.la/vwxyz'
+
+  return {
+    send (body) {
+      log.info('sms.send', {
+        phoneNumber: body.phoneNumber,
+        messageId: body.messageId,
+        acceptLanguage: body.acceptLanguage
+      })
+
+      return P.resolve()
+        .then(() => {
+          const senderId = getSenderId(body.phoneNumber)
+          const message = getMessage(body.messageId, body.acceptLanguage)
+
+          return sendSms(senderId, body.phoneNumber, message.trim())
+        })
+        .then(result => {
+          const resultCount = result.messages && result.messages.length
+          if (resultCount !== 1) {
+            log.error('sms.send', { err: new Error('Unexpected result count'), result })
+          }
+
+          result = result.messages[0]
+          const status = result.status
+
+          if (status !== '0') {
+            fail(`Delivery failed: ${status} ${result['error-text']}`, 500)
+          }
+        })
+    }
+  }
+
+  function getSenderId (phoneNumber) {
+    const numberUtil = libpn.PhoneNumberUtil.getInstance()
+    const parsedNumber = numberUtil.parse(phoneNumber)
+
+    if (! numberUtil.isValidNumber(parsedNumber)) {
```

Comment:
Or did you mean that you assume *some validation* will be done on the auth server, rather than explicitly this validation?

I do have rudimentary validation at the API level, yes. `/^\+[1-9]\d{1,14}$/` is the regex, if you're interested.